src/main/java/org/opensearch/knn/plugin/KNNPlugin.java

```diff
@@ -359,6 +374,16 @@ public Settings additionalSettings() {
             IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS.getDefault(Settings.EMPTY).stream(),
             engineSettings.stream()
         ).collect(Collectors.toList());
-        return Settings.builder().putList(IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS.getKey(), combinedSettings).build();
+
+        settingsBuilder.putList(IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS.getKey(), combinedSettings);
+    }
+
+    private Settings pluginAdditionalSettings(final Settings templateAndRequestSettings) {
+        final var settingsBuilder = Settings.builder();
+        boolean isKnnIndex = Boolean.parseBoolean(templateAndRequestSettings.get(KNNSettings.KNN_INDEX, Boolean.FALSE.toString()));
+        if (isKnnIndex) {
+            additionalMMapFileExtensionsForHybridFs(settingsBuilder);
```

Comment:
Do we ensure that if the user overrides this setting, that override gets respected? I believe we need to check templateAndRequestSettings and prioritize the user provided override

Reply:
let me check which values take priority

Reply:
testing shows that it works as expected, in particular if index has customer overwrites for settings it takes priority over values that are set by plugin. 

1. index settings: "store.hybrid.mmap.extensions" :  ["nvd", "dvd", "tim", "tip", "dim", "kdd", "kdi", "cfs", "doc", "aaa", "bbb"]
2. plugin sets ["nvd", "dvd", "tim", "tip", "dim", "kdd", "kdi", "cfs", "doc", "vec", "vex"]
3. actual values used in runtime: ["nvd", "dvd", "tim", "tip", "dim", "kdd", "kdi", "cfs", "doc", "aaa", "bbb"]